### PR TITLE
Adding referalls and requests fetch hook

### PR DESCRIPTION
**Pull request by wullaski**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- Adds data fetching and unit tests for the referrals and requests page

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
[department-of-veterans-affairs/va.gov-team#97642](https://github.com/department-of-veterans-affairs/va.gov-team/issues/97642)

## Testing done

- Unit tests
- Visual

To test navigate to `http://localhost:3001/my-health/appointments/referrals-requests`
- [x] Modify responses in `src/applications/vaos/services/mocks/index.js` for error and empty states
  - 'GET /vaos/v2/appointments': (req, res) => {
  - 'GET /vaos/v2/epsApi/referralDetails': (req, res) => {
- [x] Check that referrals and requests are stored in redux properly

## Screenshots

Referrals and Requests:
image
Requests with no pending and one cancelled:
image
Requests
Requests with no pending and no cancelled:
image
No referrals no requests:
image
Referral fetch error:
image
Requests Error:
image
Both calls fail(more likely the case):
image

## What areas of the site does it impact?

CC referrals and requests

## Acceptance criteria

- [ ] Fetches data and stores to redux when not present

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/vaos/services/mocks/index.js [hunk 1]
```diff
@@ -499,6 +499,7 @@ const responses = {
 
   // EPS api
   'GET /vaos/v2/epsApi/referralDetails': (req, res) => {
+    // return res.status(500).json({ error: true });
```
Comment (juancstlm-a6) [inline src/applications/vaos/services/mocks/index.js:502]: Nit: remove?

**Review by brianseek — CHANGES_REQUESTED** [review]
@wullaski this all looks pretty good to me. One thing is that I see a console error on the referrals and requests page of `Warning: Failed prop type: The prop referralsError is marked as required in ReferralList, but its value is null.`

File: src/applications/vaos/referral-appointments/components/ReferralList.jsx [hunk 2]
```diff
@@ -3,45 +3,64 @@ import PropTypes from 'prop-types';
...
+        );
+      })}
+    </ul>
   );
 };
 
 ReferralList.propTypes = {
-  referrals: PropTypes.arrayOf(PropTypes.string).isRequired,
+  referrals: PropTypes.arrayOf(PropTypes.object).isRequired,
+  referralsError: PropTypes.bool.isRequired,
```
Comment (brianseek) [inline src/applications/vaos/referral-appointments/components/ReferralList.jsx:63]: You have this set to required, but have a check for it as a conditional prop on line 15. This is causing a console error when there is no error.

Comment (brianseek) [reply src/applications/vaos/referral-appointments/components/ReferralList.jsx:63]: Oh never mind, I see its a bool. You need to update `const [requestsError, setRequestsError] = useState(null);` to `const [requestsError, setRequestsError] = useState(false);` in your hook

**Comment by wullaski** [comment]
@ryanshaw This PR is ready for review

**Review by brianseek — DISMISSED** [review]
Changes look good 👍

**Review by ryanshaw — DISMISSED** [review]
LGTM

**Comment by wullaski** [comment]
@ryanshaw when you got a minute, I think I need another approval I had to re-base the branch this morning

**Review by brianseek — DISMISSED** [review]
rebase looks good

**Review by ryanshaw — DISMISSED** [review]

**Review by brianseek — APPROVED** [review]
FF change looks good

**Comment by wullaski** [comment]
@ryanshaw I noticed last minute that I needed a feature flag around the route so this needs another approval, Thanks!

**Review by IGallupSoCo — APPROVED** [review]
LGTM

**Pull request merged**